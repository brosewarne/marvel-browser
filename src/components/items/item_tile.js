import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';

import { ItemImage } from '../items/item_image';

import './items.css';

/**
 * Render a single CharacterTile
 * @param {Object} item - The item JSON definition
 * @param {String} itemType - The type of item - character, comic or series
 * @param {String} imgSize - The size of image to use - eg portrait_medium, portrait_uncanny
 * @returns {Node} - The rendered CharacterTile
 * @constructor
 */
export const ItemTile = ({ item, itemType, imgSize = 'portrait_medium' }) => {
    const characterImage = <ItemImage imgSize={imgSize} item={item} />;
    const href = `/${itemType}/id/${item.id}`;
    return (
        <Paper elevation={12} className="itemTile" key={`${item.id}_tile`}>
            <h1 className="itemTitle">{item.name || item.title}</h1>
            <Link
                href={href}
                to={{ pathname: href, state: { item } }}
                title={`Click to see ${item.name}'s profile`}
            >
                {characterImage}
            </Link>
        </Paper>
    );
};

ItemTile.propTypes = {
    item: PropTypes.object.isRequired,
    itemType: PropTypes.string.isRequired,
    imgSize: PropTypes.string
};
