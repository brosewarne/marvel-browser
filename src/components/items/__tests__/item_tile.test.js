import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { ItemTile } from '../item_tile';

configure({ adapter: new Adapter() });

describe('<CharacterTile />', () => {
    const mockCharacter = {
        id: 1009146,
        name: 'Abomination',
        thumbnail: {
            path: '/a/href/to/an/image',
            extension: 'png'
        }
    };
    it('renders correctly', () => {
        const wrapper = shallow(
            <ItemTile item={mockCharacter} itemType="characters" imgSize="portrait_medium" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
