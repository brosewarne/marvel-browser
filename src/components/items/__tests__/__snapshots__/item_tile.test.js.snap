// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterTile /> renders correctly 1`] = `
<WithStyles(Paper)
  className="itemTile"
  elevation={12}
  key="1009146_tile"
>
  <h1
    className="itemTitle"
  >
    Abomination
  </h1>
  <Link
    href="/characters/id/1009146"
    replace={false}
    title="Click to see Abomination's profile"
    to={
      Object {
        "pathname": "/characters/id/1009146",
        "state": Object {
          "item": Object {
            "id": 1009146,
            "name": "Abomination",
            "thumbnail": Object {
              "extension": "png",
              "path": "/a/href/to/an/image",
            },
          },
        },
      }
    }
  >
    <Component
      imgSize="portrait_medium"
      item={
        Object {
          "id": 1009146,
          "name": "Abomination",
          "thumbnail": Object {
            "extension": "png",
            "path": "/a/href/to/an/image",
          },
        }
      }
    />
  </Link>
</WithStyles(Paper)>
`;
